runtime/runtime/src/ext.rs

```diff
@@ -381,9 +381,9 @@ impl<'a> Contract for RuntimeContractExt<'a> {
         let chain_id = self.chain_id;
         if checked_feature!("stable", EthImplicitAccounts, version)
             && account_id.get_account_type() == AccountType::EthImplicitAccount
-            && code_hash_matches_wallet_contract(chain_id, &code_hash)
+            && code_hash_matches_wallet_contract(chain_id, &code_hash, version)
```

Comment:
How does this work for accounts that have been created with the old contract? My reading of the code is that due to this condition the `wallet_contract` branch won't be taken here, because `code_hash_matches_wallet_contract("testnet", old_hash, new_protocol_version)` will return false here...?

Reply:
Furthermore we now have 3 distinct testnet contract versions with logic between the two changes differing. Could the logic between these be unified somehow?

Reply:
`code_hash_matches_wallet_contract` returns true for both [the old](https://github.com/near/nearcore/blob/322453405c627647261db3456050973b165337a8/runtime/near-wallet-contract/src/lib.rs#L51) and new contract code hashes. [There is a test for this](https://github.com/near/nearcore/blob/322453405c627647261db3456050973b165337a8/runtime/near-wallet-contract/src/lib.rs#L102).

The logic between the two versions of the wallet contract is nearly identical. There is a bug fix in the new one which covers an edge case, but otherwise the two contracts behave the same.

Reply:
But the test case that you're referencing is not testing (even after the changes in this PR) the case that I pointed out, does it?

In particular it tests a bunch of cases for `PROTOCOL_VERSION`, but my understanding that the `code_hash` variable here might be established at an older protocol version and so will remain `wallet_contract_magic_bytes(id, 70).hash()` for those older accounts, rather than `wallet_contract_magic_bytes(id, PROTOCOL_VERSION).hash()`. Is that not correct?

Reply:
Basically I'm saying that

```
assert!(code_hash_matches_wallet_contract(
    "testnet",
    wallet_contract_magic_bytes(id, 70).hash(),
    PROTOCOL_VERSION
));
```

should hold too, from what I understand. And if my understanding is wrong, I would like to understand what's the intended behaviour for those older accounts after the protocol upgrade, in that case.